esp/logging/loggingagent/espserverloggingagent/loggingagent.cpp

```diff
@@ -429,6 +436,13 @@ void CESPServerLoggingAgent::filterLogContent(IEspUpdateLogRequestWrap* req)
                 }
                 else if (group == ESPLCGUserReq)
                     originalContentTree.setown(req->getUserRequest());
+                else if (group == ESPLCGLogDatasets)
+                {
+                    const char* logDatasets = req->getLogDatasets();
+                    if (!logDatasets || !*logDatasets)
```

Comment:
Unless we're within a loop i didn't see, it seems the logic would be clearer like:
if (logDatasets && *logDatasets)
    originalContentTree.setown(createPTreeFromXMLString(logDatasets));
// no need for the continue...